core/cli_entry.py

```diff
@@ -20,25 +24,44 @@ def validate_and_upload_single_entry(directory):
     Starts the upload
 
     :param directory: Directory of the sequencing run to upload
+    :param force_upload: When set to true, the upload status file will be ignored and file will attempt to be uploaded
     :return:
     """
     logging_start_block()
     logging.debug("validate_and_upload_single_entry:Starting {}".format(directory))
 
+    # Only upload if run is new, or force_upload is True
+    if not force_upload:
+        run_is_new = parsing_handler.run_is_new(directory)
+        if not run_is_new:
+            logging.error("ERROR! Run is not new. It has either been uploaded, or an upload was attempted with error. "
+                          "Please check the status file in the directory for more details. "
+                          "You can bypass this error by uploading with the --force argument.")
+            return exit_error()
```

Comment:
My idea was that the functions in cli_entry are that top of the stack, anything that calls into cli_entry is just using the top level to do the work. Maybe it will be worth looking into changing this when a GUI is made.

Reply:
That makes sense. They are nicely organized all in one file.

I guess my comment was for something like what I do with `staramr` (https://github.com/phac-nml/staramr/blob/master/bin/staramr#L67). That is, I have a single try/except statement, which logs the error messages as `error` and returns with an `error` exit code. It could be more complicated to apply for your situation though, depending on the number of custom code you want to run in each error situation.

This was just a thought though. Could be looked into more later.